addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -91,14 +91,7 @@ var ProductComparison = publicWidget.Widget.extend(VariantMixin, {
     handleCompareAddition: function ($elem) {
         var self = this;
         if (this.comparelist_product_ids.length < this.product_compare_limit) {
-            var productId = $elem.data('product-product-id');
-            if ($elem.hasClass('o_add_compare_dyn')) {
-                productId = $elem.parent().find('.product_id').val();
-                if (!productId) { // case List View Variants
-                    productId = $elem.parent().find('input:checked').first().val();
-                }
-            }
-
+            var productId = $elem[0].dataset.productProductId;
```

Comment:
`product-product-id` seems to be overridden through the `data(` function of jQuery so reading it like this will not work.

Reply:
I can't change the way it is done ? I changed the other data access/write

Reply:
Yes you can change, but when I checked the write access were not changed

Reply:
OK Maybe I had remaining changes on my local branch...

Reply:
If this is done, resolve the conversation :)